digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/ProjectValidator.scala

```diff
@@ -745,24 +748,25 @@ object ProjectValidator {
       groupedDiscontinuity
     }
 
-    def checkEndOfRoadOutsideOfProject(project: RoadAddressProject, projectLinks: Seq[ProjectLink]): Seq[ValidationErrorDetails] = {
-      endOfRoadOutSideProject.toSeq
-    }
-
-    def endOfRoadOutSideProject: Option[ValidationErrorDetails] = {
-      val (road, part): (Long, Long) = (seq.head.roadNumber, seq.head.roadPartNumber)
-      val previousRoadPartNumber: Option[Long] = RoadAddressDAO.fetchPreviousRoadPartNumber(road, part)
-      if (previousRoadPartNumber.nonEmpty) {
-        val allProjectLinks = ProjectDAO.getProjectLinks(project.id)
-        if (allProjectLinks.exists(link => link.roadNumber == road && link.roadPartNumber == previousRoadPartNumber.get))
-          return None
-        val previousRoadPartEnd: Option[RoadAddress] = RoadAddressDAO.fetchByRoadPart(road, previousRoadPartNumber.get, fetchOnlyEnd = true).headOption
-        if (previousRoadPartEnd.nonEmpty) {
-          if (previousRoadPartEnd.get.discontinuity == EndOfRoad)
-            return outsideOfProjectError(project.id, alterMessage(ValidationErrorList.DoubleEndOfRoad, roadAndPart = Some(Seq((previousRoadPartEnd.get.roadNumber, previousRoadPartEnd.get.roadPartNumber)))))(List(previousRoadPartEnd.get))
+    def checkEndOfRoadOutsideOfProject: Seq[ValidationErrorDetails] = {
+      val (road, part): (Long, Long) = (roadProjectLinks.head.roadNumber, roadProjectLinks.head.roadPartNumber)
+      RoadAddressDAO.fetchPreviousRoadPartNumber(road, part) match {
+        case Some(previousRoadPartNumber) => if (!allProjectLinks.exists(link => link.roadNumber == road && link.roadPartNumber == previousRoadPartNumber)) {
```

Comment:
I think this is not correct for every cases.
Imagine that we have a transfer where we just change the road_part_number and we change the Tien Loppu code to another part, since we replaced the road_part_numbers. This will cause troubles. I'm fixing this problem on VIITE-1445 and 1446. But I can test on this branch

Reply:
Fixed